mqtt/mqtt-bridge/src/pump/egress.rs

```diff
@@ -59,28 +61,44 @@ where
         info!("starting egress publication processing...");
 
         let mut shutdown = shutdown_recv.fuse();
-        let mut loader = store.loader().fuse();
+
+        // Take the stream of loaded messages and convert to a stream of futures which publish.
+        // Then convert to buffered stream so that we can have multiple in-flight and also limit number of publications.
+        let loader = store.loader();
+        let load_and_publish = loader
+            .filter_map(|loaded| async {
```

Comment:
It'd be a bit easier to grasp if Ok block or whole body to be moved to a closure or separate method